tests/pods_test.go

```diff
@@ -1,16 +1,97 @@
 package tests
 
 import (
+	"context"
+	"fmt"
+	"github.com/onsi/gomega/types"
+	"strings"
+
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/k8snetworkplumbingwg/kubemacpool/pkg/names"
 )
 
+const defaultNumberOfReplicas = 2
+
 var _ = Describe("Pods", func() {
-	Context("Check the client", func() {
-		It("should not fail", func() {
-			_, err := testClient.KubeClient.CoreV1().Pods("").List(v1.ListOptions{})
+	Context("Check the pod mutating webhook", func() {
+		AfterEach(func() {
+			// Clean pods from our test namespaces after every test to start clean
+			for _, namespace := range []string{TestNamespace, OtherTestNamespace} {
+				podList := &corev1.PodList{}
+				err := testClient.VirtClient.List(context.TODO(), &client.ListOptions{Namespace: namespace}, podList)
```

Comment:
Are we sure this is the correct parameter order? in [master](https://github.com/k8snetworkplumbingwg/kubemacpool/blob/75371d2d5442ac33f56a8500f58d6bdce19805d5/tests/pods_test.go#L26) branch it's deifferent

Reply:
Interesting.

The `List` signature on this branch is `List(ctx context.Context, opts *ListOptions, list runtime.Object) error` https://github.com/k8snetworkplumbingwg/kubemacpool/blob/release-v0.8/vendor/sigs.k8s.io/controller-runtime/pkg/client/interfaces.go#L54

On master it is `List(ctx context.Context, list runtime.Object, opts ...ListOption) ` https://github.com/k8snetworkplumbingwg/kubemacpool/blob/master/vendor/sigs.k8s.io/controller-runtime/pkg/client/interfaces.go#L61

Seems that it changed between controller-runtime versions. Good catch!